src/gc/gc.cpp

```diff
@@ -2963,8 +2966,12 @@ gc_heap::dt_low_ephemeral_space_p (gc_tuning_point tp)
             
             dprintf (GTC_LOG, ("h%d: plan eph size is %Id, new gen0 is %Id", 
                 heap_number, plan_ephemeral_size, new_gen0size));
-            ret = ((size_t)(heap_segment_reserved (ephemeral_heap_segment) - (heap_segment_mem (ephemeral_heap_segment))) <
-                    (plan_ephemeral_size + new_gen0size));
+
+            // While we were in the no gc region we could have allocated a larger than normal segment,
```

Comment:
OK. As with the other comment, I think this needs to be rephrased in order to make it clear outside of the context of this PR. Can you rephrase this to something like "in cases where we were in a no-gc region..."?

"While we were in the no gc region" implies that we had to have been in a no-gc region, which is not necessarily the case.